spec/ddoc.dd

```diff
@@ -968,33 +970,65 @@ $(P
 
 $(P
     The replacement text is recursively scanned for more macros.
-    If a macro is recursively encountered, with no argument or with
+    If found, they are expanded in turn.
+    If a macro already expanded is recursively encountered, with no argument or with
     the same argument text as the enclosing macro, it is replaced
     with no text.
+)
+$(UL
+    $(LI
     Macro invocations that cut across replacement text boundaries are
     not expanded.
-    If the macro name is undefined, the replacement text has no characters
-    in it.
-    If a $(DOLLAR)(NAME) is desired to exist in the output without being
-    macro expanded, the $(DOLLAR) should be $(LINK2 #punctuation_escapes,
+    )
+    $(LI
+    If the macro name is undefined, the replacement text will
+    be $(D $)`(DDOC_UNDEFINED_MACRO(NAME))`. This defaults to empty.
+    )
+    $(LI
+    If $(D $)`(NAME)` is desired to exist in the output without being
```

Comment:
`desired` sounds a bit non-technical to me. Maybe replace with `is required in the output [...]`.

Reply:
```suggestion
    If $(D $)`(NAME)` is required to exist in the output without being
```